src/main/java/com/checkmarx/flow/service/FlowService.java

```diff
@@ -152,33 +152,56 @@ public ScanDetails executeCxScan(ScanRequest request, File cxFile) throws Machin
                     projectId = cxService.getProjectId(ownerId, projectName); //get the project id of the updated or created project
                 }
                 osaScanId = createOsaScan(request, projectId);
-                logRequest(request, osaScanId);
+                if(osaScanId != null) {
+                    logRequest(request, osaScanId, cxFile, Status.SUCCESS);
+                }
                 //resultsService.processScanResultsAsync(request, projectId, scanId, osaScanId, request.getFilters());
             }
 
 
         }catch (CheckmarxException | GitAPIException e){
             log.error(ExceptionUtils.getMessage(e), e);
             Thread.currentThread().interrupt();
-            new ScanReport(scanId, request, sourcesPath, Status.FAILURE.build(e.getMessage())).log();
+            logRequest(request, scanId, cxFile, Status.FAILURE.build(e.getMessage()));
             throw new MachinaException("Checkmarx Error Occurred");
         }
 
-        logRequest(request, scanId);
-        
+        logRequest(request, scanId, cxFile, Status.SUCCESS);
+
         this.scanDetails = new ScanDetails(projectId, scanId, osaScanId);
         return scanDetails;
     }
 
-    private void logRequest(ScanRequest request, Integer scanId)  {
-        new ScanReport(scanId, request, sourcesPath, Status.SUCCESS).log();
+
+    private void logRequest(ScanRequest request, Integer scanId, File  cxFile, Status status)  {
+        new ScanReport(scanId, request, getRepoUrl(request, cxFile), status).log();
+    }
+
+
+    private void logRequest(ScanRequest request, String scanId, File  cxFile, Status status)  {
+        new ScanReport(scanId, request, getRepoUrl(request, cxFile), status).log();
```

Comment:
Question - instead of maintain duplicated methods\constructors which the only different between them is the scan field type, why won't you take care at the beginning to parse one of them the other's type?